cubeviz/data_factories/__init__.py

```diff
@@ -2,3 +2,287 @@
 from .jwst import read_jwst_data_cube
 from .kmos import read_kmos_data_cube
 from .manga import read_manga_data_cube
+from glue.core import Data
+from glue.core.coordinates import coordinates_from_header
+from ..listener import CUBEVIZ_LAYOUT
+from ..layout import FLUX, ERROR, MASK
+
+from os.path import basename, splitext
+import yaml
+import os
+import glob
+import logging
+from astropy.io import fits
+import numpy as np
+from glue.config import data_factory
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger('cubeviz_data_configuration')
+logger.setLevel(logging.INFO)
+
+DEFAULT_DATA_CONFIGS = os.path.join(os.path.dirname(__file__), 'configurations')
+CUBEVIZ_DATA_CONFIGS = 'CUBEVIZ_DATA_CONFIGS'
+
+class DataConfiguration:
+    """
+    This class is used to parse a YAML configuration file.
+
+    """
+
+    def __init__(self, config_file):
+        """
+        Given the configuration file, save it and grab the name and priority
+        :param config_file:
+        """
+        self._config_file = config_file
+
+        with open(self._config_file, 'r') as ymlfile:
+            cfg = yaml.load(ymlfile)
+
+            self._name = cfg['name']
+            self._type = cfg['type']
+
+            try:
+                self._priority = int(cfg.get('priority', 0))
+            except:
+                self._priority = 0
+
+            self._configuration = cfg['match']
+
+            self._data = cfg.get('data', None)
+
+
+    def load_data(self, data_filename):
+        """
+        Load the data based on the extensions defined in the matching YAML file.  THen
+        create the datacube and return it.
+
+        :param data_filename:
+        :return:
+        """
+        hdulist = fits.open(data_filename)
+
+        try:
+            flux_index = int(self._data['FLUX'])
+        except:
+            flux_index = self._data['FLUX']
+
+        flux = hdulist[flux_index]
+        flux_data = flux.data
+
+        try:
+            error_index = int(self._data['ERROR'])
+        except:
+            error_index = self._data['ERROR']
+        var_data = hdulist[error_index].data
+
+        if not self._data['DQ'] == 'None':
+            mask_data = hdulist[self._data['DQ']].data
+        else:
+            mask_data = np.empty(flux_data.shape)
+
+        label = "{}: {}".format(self._name, splitext(basename(data_filename))[0])
+
+        data = Data(label=label)
+
+        data.coords = coordinates_from_header(flux.header)
+        data.meta[CUBEVIZ_LAYOUT] = self._name
```

Comment:
Admittedly this is missing from the original code, but it would be useful to have a comment here about how this attribute is used to indicate to the `cubeviz` layout that this is a cubeviz-specific data component.

Reply:
I added a comment in there.